internal/controller/authenticator.go

```diff
@@ -181,11 +192,23 @@ func SetUpOAuthConnection(
 	}
 	tokenSource = GetTokenSource(ctx, oauthConfig, tlsConfig)
 	creds := credentials.NewTLS(tlsConfig)
+
+	proxyDialer := func(ctx context.Context, addr string) (net.Conn, error) {
+		proxyURL, err := http.ProxyFromEnvironment(&http.Request{URL: &url.URL{Host: addr}})
+		if err != nil || proxyURL == nil {
+			logger.Warn("[Proxy Failure] No valid proxy found or error occurred", zap.Error(err))
+			return net.Dial("tcp", addr)
+		}
```

Comment:
1. In the common case, where no HTTPS_PROXY environment variable is set, this will always print this scary Warn level message for no reason. You should only print a Warn level message if the variable is set and the proxy can't be found.
1. Remove the fancy formatting and keep it simple. When you want to add structure to log entries, use fields.
```suggestion
		proxyURL, err := http.ProxyFromEnvironment(&http.Request{URL: &url.URL{Host: addr}})
		if err != nil {
			logger.Warn("Invalid HTTPS proxy configured; ignoring proxy settings", zap.Error(err))
			return net.Dial("tcp", addr)
		}
		if proxyURL == nil { // No proxy configured
			return net.Dial("tcp", addr)
		}
```